I tried to put this one on top of the PR I'm preparing and I have a test failure:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project quarkus-integration-test-keycloak-authorization: Failed to build a runnable JAR: Failed to augment application classes: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.keycloak.pep.deployment.KeycloakPolicyEnforcerBuildStep#requireBody threw an exception: java.lang.NullPointerException
[ERROR] 	at io.quarkus.keycloak.pep.deployment.KeycloakPolicyEnforcerBuildStep.isBodyClaimInformationPointDefined(KeycloakPolicyEnforcerBuildStep.java:47)
[ERROR] 	at io.quarkus.keycloak.pep.deployment.KeycloakPolicyEnforcerBuildStep.requireBody(KeycloakPolicyEnforcerBuildStep.java:30)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1395)
[ERROR] 	at java.lang.Thread.run(Thread.java:748)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java
@@ -459,9 +459,9 @@ public final class BuildTimeConfigurationReader {
                 ni.previous();
                 // now the cursor is before our map key and after the enclosing map key
                 Map<String, Object> map = getMap((MapContainer) parent, ni);
+                String key = ni.getPreviousSegment();
                 ni.next();
                 // cursor restored
-                String key = ni.getPreviousSegment();
                 Map<String, Object> instance = getAsMap(map, key);
                 if (instance == null) {
                     instance = new HashMap<>();

--- a/core/test-extension/deployment/src/main/java/io/quarkus/extest/deployment/TestProcessor.java
+++ b/core/test-extension/deployment/src/main/java/io/quarkus/extest/deployment/TestProcessor.java
@@ -20,6 +20,7 @@ import java.util.Base64;
 import java.util.Collection;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 import java.util.function.BooleanSupplier;
 
@@ -456,6 +457,16 @@ public final class TestProcessor {
                 .finalFieldsWritable(true)
                 .build();
         classes.produce(finalField);
+    }
+
+    @BuildStep
+    void checkMapMap(TestBuildAndRunTimeConfig btrt, TestBuildTimeConfig bt, BuildProducer<ReflectiveClassBuildItem> unused) {
+        if (!Objects.equals("1234", btrt.mapMap.get("outer-key").get("inner-key"))) {
+            throw new AssertionError("BTRT map map failed");
+        }
+        if (!Objects.equals("1234", bt.mapMap.get("outer-key").get("inner-key"))) {
+            throw new AssertionError("BT map map failed");
+        }
     }
 
     @BuildStep(onlyIf = Never.class)

--- a/core/test-extension/deployment/src/main/resources/application.properties
+++ b/core/test-extension/deployment/src/main/resources/application.properties
@@ -113,3 +113,8 @@ quarkus.btrt.my-enums=optional,enum-one,enum-two
 
 ### anonymous root property
 quarkus.test-property=foo
+
+### map of map of strings
+quarkus.rt.map-map.outer-key.inner-key=1234
+quarkus.btrt.map-map.outer-key.inner-key=1234
+quarkus.bt.map-map.outer-key.inner-key=1234

--- a/core/test-extension/deployment/src/test/java/io/quarkus/extest/ConfiguredBeanTest.java
+++ b/core/test-extension/deployment/src/test/java/io/quarkus/extest/ConfiguredBeanTest.java
@@ -268,4 +268,26 @@ public class ConfiguredBeanTest {
         List<Integer> actualMapValues = new ArrayList<>(configuredBean.getRunTimeConfig().mapOfNumbers.values());
         Assertions.assertEquals(mapValues, actualMapValues);
     }
+
+    @Test
+    public void testBtrtMapOfMap() {
+        Map<String, Map<String, String>> mapMap = configuredBean.getBuildTimeConfig().mapMap;
+        Assertions.assertFalse(mapMap.containsKey("inner-key"));
+        Assertions.assertTrue(mapMap.containsKey("outer-key"));
+        Map<String, String> map = mapMap.get("outer-key");
+        Assertions.assertTrue(map.containsKey("inner-key"));
+        Assertions.assertFalse(map.containsKey("outer-key"));
+        Assertions.assertEquals("1234", map.get("inner-key"));
+    }
+
+    @Test
+    public void testRtMapOfMap() {
+        Map<String, Map<String, String>> mapMap = configuredBean.getRunTimeConfig().mapMap;
+        Assertions.assertFalse(mapMap.containsKey("inner-key"));
+        Assertions.assertTrue(mapMap.containsKey("outer-key"));
+        Map<String, String> map = mapMap.get("outer-key");
+        Assertions.assertTrue(map.containsKey("inner-key"));
+        Assertions.assertFalse(map.containsKey("outer-key"));
+        Assertions.assertEquals("1234", map.get("inner-key"));
+    }
 }

--- a/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/MapMapConfig.java
+++ b/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/MapMapConfig.java
@@ -1,0 +1,17 @@
+package io.quarkus.extest.runtime.config;
+
+import java.util.Map;
+
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ *
+ */
+@ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED, name = "mm-root")
+public class MapMapConfig {
+    //### map of map of strings
+    //quarkus.mm-root.map.inner-key.outer-key=1234
+
+    Map<String, Map<String, String>> map;
+}

--- a/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/TestBuildAndRunTimeConfig.java
+++ b/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/TestBuildAndRunTimeConfig.java
@@ -38,6 +38,8 @@ public class TestBuildAndRunTimeConfig {
     @ConvertWith(WholeNumberConverter.class)
     public Map<String, Integer> mapOfNumbers;
 
+    public Map<String, Map<String, String>> mapMap;
+
     /**
      * Enum object
      */

--- a/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/TestBuildTimeConfig.java
+++ b/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/TestBuildTimeConfig.java
@@ -1,5 +1,7 @@
 package io.quarkus.extest.runtime.config;
 
+import java.util.Map;
+
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
@@ -24,6 +26,8 @@ public class TestBuildTimeConfig {
     /** A nested config group with all supported value types */
     @ConfigItem
     public AllValuesConfig allValues;
+
+    public Map<String, Map<String, String>> mapMap;
 
     public TestBuildTimeConfig() {
 

--- a/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/TestRunTimeConfig.java
+++ b/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/TestRunTimeConfig.java
@@ -107,6 +107,8 @@ public class TestRunTimeConfig {
     @ConvertWith(WholeNumberConverter.class)
     public Map<String, Integer> mapOfNumbers;
 
+    public Map<String, Map<String, String>> mapMap;
+
     @Override
     public String toString() {
         return "TestRunTimeConfig{" +

--- a/extensions/keycloak-authorization/deployment/src/main/java/io/quarkus/keycloak/pep/deployment/KeycloakPolicyEnforcerBuildStep.java
+++ b/extensions/keycloak-authorization/deployment/src/main/java/io/quarkus/keycloak/pep/deployment/KeycloakPolicyEnforcerBuildStep.java
@@ -44,8 +44,10 @@ public class KeycloakPolicyEnforcerBuildStep {
         for (Map.Entry<String, Map<String, String>> entry : claims.entrySet()) {
             Map<String, String> value = entry.getValue();
 
-            if (value.get(entry.getKey()).contains("request.body")) {
-                return true;
+            for (String nestedValue : value.values()) {
+                if (nestedValue.contains("request.body")) {
+                    return true;
+                }
             }
         }
 

--- a/integration-tests/test-extension/src/main/resources/application.properties
+++ b/integration-tests/test-extension/src/main/resources/application.properties
@@ -104,3 +104,8 @@ quarkus.btrt.my-enums=optional,enum-one,enum-two
 
 ### anonymous root property
 quarkus.test-property=foo
+
+### map of map of strings
+quarkus.rt.map-map.outer-key.inner-key=1234
+quarkus.btrt.map-map.outer-key.inner-key=1234
+quarkus.bt.map-map.outer-key.inner-key=1234

